File: Algorithms/src/ywy/java/level1/MinimumRectangle.java
package ywy.java.level1;

public class MinimumRectangle {

	public MinimumRectangle() {
	}

	public int solution(int[][] sizes) {
		int index = 0;
		int range = sizes.length;
		int[] ys = new int[range];
		int max = 0;
		int x = -1;
		int y = -1;
		do {
			x = sizes[index][0];
			y = sizes[index][1];
			ys[index] = y;
			if (y > max) {
				max = y;
			}
			if (x > max) {
				max = x;
			}
		} while (++index < range);

		index = 0;
		do {
			x = sizes[index][0];
			y = ys[index];
			if (max - y < max - x) {
				ys[index] = x;
			}
		} while (++index < range);
		return max * getMax(ys);
	}

	private int getMax(int[] arr) {
		int index = 0;
		int max = 0;
		do {
			if (max < arr[index]) {
				max = arr[index];
			}
		} while (++index < arr.length);
		return max;
	}
}

File: Algorithms/src/ywy/javascript/level1/MinimumRectangle.js
const solution = (sizes) => {
    const max = Math.max(...sizes.flat());
    const ys = [];
    sizes.forEach(el => {
        if ( max - el[0] > max - el[1]) {
            ys.push(el[0]);
            return;
        }
        ys.push(el[1]);
    });
    return Math.max(...ys) * max;
};

solution([[10, 7], [12, 3], [8, 15], [14, 7], [5, 15]]);
solution([[60, 50], [30, 70], [60, 30], [80, 40]]);
solution([[14, 4], [19, 6], [6, 16], [18, 7], [7, 11]]);